### PR TITLE
Random Battle: Split Castform into Sunny and Rainy Formes

### PR DESCRIPTION
**Pull request by ascriptmaster**
Also implement fix for Meloetta starting out in Pirouette Forme in Random
Doubles Battles.

### CODE REVIEW

File: data/scripts.js [hunk 1]
```diff
@@ -902,6 +902,11 @@ exports.BattleScripts = {
 		var baseTemplate = (template = this.getTemplate(template));
 		var name = template.name;
 
+		if (name === 'Castform') {
+			// If someone wants a Castform set, give it one of its subforme sets instead
+			template = this.getTemplate(this.random(2) ? 'castformsunny' : 'castformrainy');
```
Comment (TheImmortal) [inline data/scripts.js]: This is unnecessary. It skips Castform because it has no moves. The formes are obtainable by themselves as they now have moves.

Comment (TheImmortal) [inline data/scripts.js]: Castform can't be selected anymore as is doesn't have moves. So, this is unnecessary.

File: data/scripts.js [hunk 2]
```diff
@@ -911,6 +916,11 @@ exports.BattleScripts = {
 			require('../crashlogger.js')(fakeErr, 'The randbat set generator');
 		}
 
+		// Castform-Sunny and Castform-Rainy can be chosen
+		if (template.num === 351) {
+			name = 'Castform';
+		}
+
```
Comment (TheImmortal) [inline data/scripts.js]: No need for blank line imo.

File: data/formats-data.js [hunk 3]
```diff
@@ -2845,10 +2845,15 @@ exports.BattleFormatsData = {
 		tier: "UU"
 	},
 	castform: {
-		randomBattleMoves: ["sunnyday","raindance","thunderbolt","icebeam","weatherball"],
-		randomDoubleBattleMoves: ["sunnyday","raindance","thunderbolt","icebeam","weatherball","protect"],
+		// Random Battle Moves implemented in its subformes and the randomSet functions
```
Comment (TheImmortal) [inline data/formats-data.js]: No need for this comment imo.

File: data/scripts.js (hunk 1 above)
Comment (ascriptmaster) [inline data/scripts.js]: Seasonals though.

Comment (TheImmortal) [inline data/scripts.js]: This is the Random Battle script. We do not care about Seasonals.

Comment (ascriptmaster) [inline data/scripts.js]: `randomSet` is occasionally called by Seasonals Teambuilding scripts. This isn't `randomTeam`.

Comment (TheImmortal) [inline data/scripts.js]: The Seasonals script should take these stuff into account.

**Comment by ascriptmaster** [comment]
All right, fair enough. Do the sets look good?

**Comment by ascriptmaster** [comment]
Blah, I'll need to do some tier things in `formats-data.js` with Castform as well, won't I

**Comment by TheImmortal** [comment]
No, I don't think you do.

Ice Beam is better on Sunny than Thunderbolt because SolarBeam hits Water already.

**Comment by ascriptmaster** [comment]
I was using Tbolt as a neutral move to hit Fire with a little bit, honestly. If you think Ice Beam is better then I'll change it when I get home.

**Comment by TheImmortal** [comment]
Neutral is gonna be very weak. I almost wanted to suggest Ice Beam for Rainy so that it has BoltBeam but that is extremely weak without STAB. So yeah, I'd say just switch Bolt for Beam on Sun.

**Comment by ascriptmaster** [comment]
Right, because Fire/Grass is ALSO walled by Dragon, which also walls Electric but is weak to Ice. All right, I'll fix it when I get home.

**Comment by TheImmortal** [comment]
Also, if you could, remove https://github.com/Zarel/Pokemon-Showdown/blob/master/data/scripts.js#L1347 and add `case 'sunnyday':` to https://github.com/Zarel/Pokemon-Showdown/blob/master/data/scripts.js#L1331 and remove the check for `hasMove['sunnyday']` in that. Rain Dance + Sunny Day was only relevant to Castform.

**Comment by ascriptmaster** [comment]
So, like this.

**Comment by ascriptmaster** [comment]
Also I was considering Fire Blast/Hydro Pump for the Castform formes, but Castform without Weather Ball seems just... wrong

**Comment by TheImmortal** [comment]
I meant 08f108aefdfa487f4e182cc61061aea29bb90cbb.

**Pull request merged**